### PR TITLE
switch pause to docker manifest instead of manifest-tool

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

It wasn't possible to push pause 3.2 due to severely outdated manifest tool.

Initially I updated the manifest tool and rewrote the target to supply credentials to the manifest tool, but this is difficult to do generically across registries and many of our targets use `docker manifest ..` instead already.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: build/pause/Makefile [hunk 1]
```diff
@@ -60,8 +58,10 @@ all: all-container
 
 all-push: all-push-images push-manifest
 
-push-manifest: manifest-tool
-	manifest-tool push from-args --platforms $(call join_platforms,$(ALL_ARCH)) --template $(IMAGE)-ARCH:$(TAG) --target $(IMAGE):$(TAG)
+push-manifest:
+	docker manifest create --amend $(IMAGE):$(TAG) $(shell echo $(ALL_ARCH) | sed -e "s~[^ ]*~$(IMAGE)\-&:$(TAG)~g")
+	set -x; for arch in $(ALL_ARCH); do docker manifest annotate --arch $${arch} ${IMAGE}:${TAG} ${IMAGE}-$${arch}:${TAG}; done
```
Comment (BenTheElder) [inline build/pause/Makefile]: we usually have this line as `@` but then you can't see the manifest actually being built, instead I removed that and added trace mode so you can see the annotations ...

Comment (BenTheElder) [reply build/pause/Makefile]: this is otherwise from build/debian-base

File: test/images/Makefile [hunk 2]
```diff
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-include  ../../hack/make-rules/Makefile.manifest
```
Comment (BenTheElder) [inline test/images/Makefile]: - we don't use any rules from this
- if you inspect the targets we do have, we're using `docker manifest ...`, not `manifest-tool`

File: hack/make-rules/Makefile.manifest [hunk 3]
```diff
@@ -1,30 +0,0 @@
-# Copyright 2018 The Kubernetes Authors.
```
Comment (BenTheElder) [inline hack/make-rules/Makefile.manifest]: The pause image was the last actual consumer of this `Makefile.manifest` no longer has an results in this repo.

**Comment by BenTheElder** [comment]
cc @justaugustus @mkumatag

File: build/pause/Makefile (hunk 1 above)
Comment (cblecker) [reply build/pause/Makefile]: Is this rule standardized? Should this makerule go into `../../hack/make-rules/Makefile.manifest`?

Comment (BenTheElder) [reply build/pause/Makefile]: it's used by a few things.

I'd somewhat prefer to finish getting the pause update sorted before refactoring further but I _think_ it should be pretty simple to standardize a few makefiles on this.

**Pull request merged**